substrate/frame/election-provider-multi-block/src/signed/mod.rs

```diff
@@ -119,18 +119,48 @@ pub struct SubmissionMetadata<T: Config> {
 impl<T: Config> SolutionDataProvider for Pallet<T> {
 	type Solution = SolutionOf<T::MinerConfig>;
 
-	fn get_page(page: PageIndex) -> Option<Self::Solution> {
-		// note: a non-existing page will still be treated as merely an empty page. This could be
-		// re-considered.
+	fn get_page(page: PageIndex) -> Self::Solution {
 		let current_round = Self::current_round();
-		Submissions::<T>::leader(current_round).map(|(who, _score)| {
-			sublog!(info, "signed", "returning page {} of {:?}'s submission as leader.", page, who);
-			Submissions::<T>::get_page_of(current_round, &who, page).unwrap_or_default()
-		})
+		Submissions::<T>::leader(current_round)
+			// Leader is verified to exist before we call `Verifier::start`. The verifier only
+			// transitions to `Status::Ongoing` when a leader is confirmed to exist. During
+			// verification, the leader should remain unchanged - it's only removed when
+			// verification fails (which immediately stops the verifier) or completes successfully.
+			// Therefore, `get_page` should only be called when a leader exists.
+			.defensive()
+			.and_then(|(who, _score)| {
+				sublog!(
+					info,
+					"signed",
+					"returning page {} of {:?}'s submission as leader.",
+					page,
+					who
+				);
+				Submissions::<T>::get_page_of(current_round, &who, page)
+			})
+			.unwrap_or_default()
 	}
 
-	fn get_score() -> Option<ElectionScore> {
-		Submissions::<T>::leader(Self::current_round()).map(|(_who, score)| score)
+	fn get_score() -> ElectionScore {
+		let current_round = Self::current_round();
+		Submissions::<T>::leader(current_round)
+			// Leader is verified to exist before we call `Verifier::start`. The verifier only
```

Comment:
I would move these to the function doc (stated as `//`, not `///`, because it is not an API doc, but rather impl note) on top and state it once.